server/lib/tuist/qa/agent.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Tuist.QA.Agent do
+  @moduledoc """
+  Tuist QA agent module.
+  """
+
+  alias Tuist.QA.SimulatorController
+  alias Tuist.Zip
+
+  def test(%{preview_url: preview_url, bundle_identifier: bundle_identifier}) do
+    run_preview(preview_url, bundle_identifier)
+  end
+
+  defp run_preview(preview_url, bundle_identifier) do
```

Comment:
It might be useful to add some logs here using `Logger` such that we can pipe them through some logging platform (e.g. Grafana) and debug issues when they arise.

Reply:
I'd consider including some ID to connect the logs with some server-side testing session.

Reply:
yes, certainly, but we first need to connect the agent to the server since this will be a CLI running in the namespace environment ... let's keep that down the road, but we definitely need some error reporting of the CLI